docs/cheat-sheets/javascript-command-injection.mdx

```diff
@@ -0,0 +1,89 @@
+---
+slug: javascript-command-injection
+---
+
+import LinkToRegistryRule from "/src/components/LinkToRegistryRule"
+import CommandInjectionIntro from "/src/components/concept/_command-injection-intro.mdx"
+
+# Command injection prevention for JavaScript
+
+<CommandInjectionIntro />
+
+#### Check your project for these conditions:
+
+```bash
+semgrep --config p/javascript-command-injection
+```
+
+---
+
+## 1. Running an OS command
+
+### 1.A. Spawning a shell with exec function
+
+`exec()` spawns a shell then executes the command within that shell, buffering any generated output. Never pass unsanitized user input to this function. Any input containing shell metacharacters can be used to trigger arbitrary command execution. 
+
+Example:
+
+```javascript
+child_process.exec(...)
+child_process.execSync(...)
+```
+
+#### References
+
+- [`child_process` module documentation](https://nodejs.org/dist/latest-v14.x/docs/api/child_process.html)
+
+#### Mitigation
+
+Try to avoid non-literal values for the command string. If it is not possible, then do not let running arbitrary commands, use a white list for inputs. Strip `!@#$;&*~"'{}][-+%^` characters from an input provided for command string. 
```

Comment:
```suggestion
Try to avoid non-literal values for the command string. If it is not possible, then do not let running arbitrary commands, use an allow list for inputs. Strip `!@#$;&*~"'{}][-+%^` characters from an input provided for the command string. 
```